easybuild/easyblocks/b/blender.py

```diff
@@ -0,0 +1,113 @@
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Blender, implemented as an easyblock
+
+@author: Samuel Moors (Vrije Universiteit Brussel)
+"""
+import glob
+import os
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_Blender(CMakeMake):
+    """Support for building Blender."""
+
+    def find_glob_pattern(self, glob_pattern):
+        if self.dry_run:
+            return glob_pattern
+        res = glob.glob(glob_pattern)
+        if len(res) != 1:
+            raise EasyBuildError("Was expecting exactly one match for '%s', found %d: %s", glob_pattern, len(res), res)
+        return res[0]
+
+    def configure_step(self):
+        """Set CMake options for Blender"""
+        self.cfg['separate_build_dir'] = True
+
+        default_config_opts = {
+            'WITH_INSTALL_PORTABLE': 'OFF',
+            'WITH_BUILDINFO': 'OFF',
+            # disable SSE detection to give EasyBuild full control over optimization compiler flags being used
+            'WITH_CPU_SSE': 'OFF',
+            'CMAKE_C_FLAGS_RELEASE': '-DNDEBUG',
+            'CMAKE_CXX_FLAGS_RELEASE': '-DNDEBUG',
+            # these are needed unless extra dependencies are added for them to work
+            'WITH_GAMEENGINE': 'OFF',
+            'WITH_SYSTEM_GLEW': 'OFF',
+        }
+        for key in default_config_opts:
+            opt = '-D%s=' % key
+            if opt not in self.cfg['configopts']:
+                self.cfg.update('configopts', opt + default_config_opts[key])
+
+        # Python paths
+        python_root = get_software_root('Python')
+        if python_root:
+            shlib_ext = get_shared_lib_ext()
+            pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+            site_packages = os.path.join(python_root, 'lib', 'python%s' % pyshortver, 'site-packages')
+
+            numpy_root = self.find_glob_pattern(
```

Comment:
@smoors Maybe include a comment here that we assume that `numpy/requests` are included with the Python installation.

Note that this will need to be updated for the `2019a` generation where `numpy` lives in `SciPy-bundle` (but this can be tackled in a follow-up PR, not need to block this one for it imho).